packages/terminal/src/browser/terminal-widget-impl.ts

```diff
@@ -310,11 +322,11 @@ export class TerminalWidgetImpl extends TerminalWidget implements StatefulWidget
      * If id is provided attach to the terminal for this id.
      */
     async start(id?: number): Promise<number> {
-        this.terminalId = typeof id !== 'number' ? await this.createTerminal() : await this.attachTerminal(id);
+        this._terminalId = typeof id !== 'number' ? await this.createTerminal() : await this.attachTerminal(id);
         this.resizeTerminalProcess();
         this.connectTerminalProcess();
+        this.onDidOpenEmitter.fire(undefined);
         if (IBaseTerminalServer.validateId(this.terminalId)) {
-            this.onDidOpenEmitter.fire(undefined);
             return this.terminalId;
         }
         throw new Error('Failed to start terminal' + (id ? ` for id: ${id}.` : '.'));
```

Comment:
let's fire a new event here `onDidOpenFailure`, you will need to wait for it the task widget manager